js/apps/events-page/controllers/eventsPageCtrl.js

```diff
@@ -1,39 +1,212 @@
-			requestModel.StartDate = dateUtility.addDays(requestModel.StartDate, 1);
-			requestModel.EndDate = dateUtility.addDays(requestModel.EndDate, 1);
+			requestModel.Page += 1;
 
-			eventsService.get(requestModel)
-				.then((eventGroups) => {
-					self.eventGroups = self.eventGroups.concat(eventGroups);
+			eventsService.get(requestModel).then(processAndCombineEvents).then(() => {
+				$timeout(() => {
+					$('.event-date-bar').sticky(eventDateBarStickySettings);
 				});
+			});
 		};
 
-		/* ** Init ** */
+		self.clearFilters = () => {
+			requestModel.StartDate = startDateLocaleString;
+			requestModel.EndDate = endDateLocaleString;
+			requestModel.Page = 1;
+			requestModel.Keyword = '';
+			requestModel.AgeGroups = [];
+			requestModel.EventsTypes = [];
+			requestModel.Locations = [];
+
+			self.keywords = '';
+			self.userStartDate = '';
+			self.userEndDate = '';
+			self.eventGroups = [];
+			self.hasResults = true;
+			self.isLoading = true;
+			self.locations = requestModel.Locations;
+			self.eventsTypes = requestModel.EventsTypes;
+			self.ageGroups = requestModel.AgeGroups;
+	
+			eventsService.get(requestModel).then(processEvents);
+		};
 
-		eventsService.get(requestModel)
-			.then((eventGroups) => {
-				self.eventGroups = eventGroups;
+		/* ** Private ** */
+
+		const processEvents = (eventResults) => {
+			self.isLastPage = isLastPage(eventResults.totalResults);
+			self.eventGroups = eventResults.eventGroups;
+			self.isLoading = false;
+			self.hasResults = eventResults.eventGroups.length;
+
+			$timeout(() => {
+				$('.event-date-bar').sticky(eventDateBarStickySettings);
 			});
+		};
+
+		const processAndCombineEvents = (eventResults) => {
+			self.isLastPage = isLastPage(eventResults.totalResults);
+			self.eventGroups = combineEventGroups(self.eventGroups, eventResults.eventGroups);
+		};
+
+		const isLastPage = (totalResults) => {
+			const totalResultsSoFar = requestModel.Page * self.chunkSize;
+			return totalResultsSoFar >= totalResults;
+		};
+
+		const isDateRangeValid = (firstDate, secondDate) => {
+			if (firstDate && secondDate) {
+				return moment(firstDate).isSameOrBefore(secondDate);
+			}
+
+			return false;
+		};
+
+		const isSameDay = (day1Date, day2Date) => {
+			if (day1Date && day2Date) {
+				return moment(day1Date).isSame(day2Date, 'day');
+			}
+
+			return false;
+		};
+
+		const combineEventGroups = (oldEventGroups, newEventGroups) => {
+			let renderedEventGroups = oldEventGroups;
+			let lastEventGroup = renderedEventGroups[renderedEventGroups.length - 1];
+
+			angular.forEach(newEventGroups, (eventGroup) => {
+				if (isSameDay(lastEventGroup.date, eventGroup.date)) {
+					lastEventGroup.events = lastEventGroup.events.concat(eventGroup.events);
+				} else {
+					renderedEventGroups.push(eventGroup);
+				}
+			});
+
+			return renderedEventGroups;
+		};
+
+		const showFilters = (collapseEvent) => {
+			const $collapsible = angular.element(collapseEvent.currentTarget);
```

Comment:
Could these two methods be combined into a toggle filters method? Could this functionality be included in the template using ng-class?

Reply:
Yes, but it turns out I can remove the while thing since the filters no longer collapse.